src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -3667,4 +3667,576 @@ void CodeGen::genSIMDSplitReturn(GenTree* src, ReturnTypeDesc* retTypeDesc)
 }
 #endif // FEATURE_SIMD
 
+//------------------------------------------------------------------------
+// genPushCalleeSavedRegisters: Push any callee-saved registers we have used.
+//
+// Arguments (arm64):
+//    initReg        - A scratch register (that gets set to zero on some platforms).
+//    pInitRegZeroed - OUT parameter. *pInitRegZeroed is set to 'true' if this method sets initReg register to zero,
+//                     'false' if initReg was set to a non-zero value, and left unchanged if initReg was not touched.
+//
+#if defined(TARGET_ARM64)
+void CodeGen::genPushCalleeSavedRegisters(regNumber initReg, bool* pInitRegZeroed)
+#else
+void CodeGen::genPushCalleeSavedRegisters()
+#endif
+{
+    assert(compiler->compGeneratingProlog);
+
+#ifdef TARGET_ARM64
+    // Probe large frames now, if necessary, since genPushCalleeSavedRegisters() will allocate the frame. Note that
+    // for arm64, genAllocLclFrame only probes the frame; it does not actually allocate it (it does not change SP).
+    // For arm64, we are probing the frame before the callee-saved registers are saved. The 'initReg' might have
+    // been calculated to be one of the callee-saved registers (say, if all the integer argument registers are
+    // in use, and perhaps with other conditions being satisfied). This is ok in other cases, after the callee-saved
+    // registers have been saved. So instead of letting genAllocLclFrame use initReg as a temporary register,
+    // always use REG_SCRATCH. We don't care if it trashes it, so ignore the initRegZeroed output argument.
+    bool ignoreInitRegZeroed = false;
+    genAllocLclFrame(compiler->compLclFrameSize, REG_SCRATCH, &ignoreInitRegZeroed,
```

Comment:
Seems a little weird to put the call to do probing inside of a function that's supposed to push callee saved registers, but ok.

Reply:
Yeah, agree. On other hand, `genAllocLclFrame` is not allocating anything on Arm64 and should be removed as soon as I replace the stack probing logic with helper call/or *"probe while adjusting sp"* algorithm as we discussed earlier.

Reply:
In fact, I think that `genPushCalleeSavedRegisters` should be renamed as `genAllocLclFrameAndStoreCalleeSavedRegisters`.